src/views/product-integrations-landing/contexts/integrations-search-context/index.tsx

```diff
@@ -0,0 +1,325 @@
+			(integration: Integration) => {
+				let tierMatch = true
+				if (queryParams.tiers.length > 0) {
+					tierMatch = queryParams.tiers.includes(integration.tier)
+				}
+
+				let componentMatch = true
+				if (queryParams.components.length > 0) {
+					componentMatch = integration.components.some(
+						(component: IntegrationComponent) => {
+							return queryParams.components.includes(component.slug)
+						}
+					)
+				}
+
+				let flagMatch = true
+				if (queryParams.flags.length > 0) {
+					flagMatch = integration.flags.some((flag: Flag) => {
+						return queryParams.flags.includes(flag.slug)
+					})
+				}
+
+				let filterQueryMatch = true
+				if (queryParams.filterQuery.length) {
+					filterQueryMatch = getDoesMatchFilterQuery({
+						integration,
+						filterQuery: queryParams.filterQuery,
+					})
+				}
+
+				return tierMatch && componentMatch && flagMatch && filterQueryMatch
+			}
+		)
+
+		/**
+		 * Second, paginate the filtered list
+		 */
+		const paginatedIntegrations = filteredIntegrations.slice(
+			(page - 1) * pageSize,
+			page * pageSize
+		)
+
+		/**
+		 * Third, update the `hasFilteredAfterRouterReady` ref to indicate that
+		 * filtering has been completed.
+		 */
+		if (hasFilteredAfterRouterReady.current === false) {
+			hasFilteredAfterRouterReady.current = true
+		}
+
+		/**
+		 * Lastly, return the data as a neatly packaged object.
+		 */
+		return {
+			atLeastOneFacetSelected,
+			filteredIntegrations,
+			paginatedIntegrations,
+		}
+	}, [integrations, page, pageSize, queryParams, router.isReady])
+
+	/**
+	 * Generate the options arrays for UI elements.
+	 *
+	 * @TODO move this to the component that actually renders UI?
+	 */
+	const { componentOptions, flagOptions, tierOptions } = useMemo(() => {
+		return {
+			componentOptions: allComponents.map((component: IntegrationComponent) => {
+				return {
+					id: component.slug,
+					label: capitalize(component.plural_name),
+					onChange: () => {
+						resetPage()
+						toggleComponentChecked(component)
+					},
+					selected: queryParams.components.includes(component.slug),
+				}
+			}),
+			flagOptions: allFlags.map((flag: Flag) => {
+				return {
+					id: flag.slug,
+					label: flag.name,
+					onChange: () => {
+						resetPage()
+						toggleFlagChecked(flag)
+					},
+					selected: queryParams.flags.includes(flag.slug),
+				}
+			}),
+			tierOptions: allTiers.map((tier: Tier) => {
+				return {
+					id: tier,
+					label: capitalize(tier),
+					onChange: () => {
+						resetPage()
+						toggleTierChecked(tier)
+					},
+					selected: queryParams.tiers.includes(tier),
+				}
+			}),
+		}
+	}, [
+		allComponents,
+		allFlags,
+		allTiers,
+		queryParams,
+		resetPage,
+		toggleComponentChecked,
+		toggleFlagChecked,
+		toggleTierChecked,
+	])
+
+	/**
+	 * If `isLoading` hasn't already been updated, and it's ready to be updated,
+	 * update it after a 1.5 second delay. The purpose of the delay is to
```

Comment:
nit: I don't see the the arbitrary timeout any longer, we should update the comment here if we aren't planning to add that back.